Question link

terichadbourne opened the pull request:
Added a message to the bottom of the lesson template that encourages users to submit their questions/feedback on the lesson, linking to a new issue with the lesson number / title / path pre-filled and a body that explains the type of feedback we'd appreciate. 

Addresses issues #45 and #42.

olizilla commented:
@terichadbourne thanks! Pls can you drop a screenshot of what you expect to see on any PR with a visual change, so when we run it, we know that we're on the same page.

terichadbourne commented:
Thanks for the great feedback, @olizilla. I'll work on those updates. 

Here's the visual change expected, the addition of this sentence/link under the exercise box on every lesson page on the site: 

![image](https://user-images.githubusercontent.com/19171465/47036711-09aec080-d14b-11e8-82ff-c6f69c43d33c.png)

terichadbourne commented:
My latest commit to this branch should address all the issues that @olizilla flagged, as well as adding encoded formatting to the body of the new issue. 

@mikeal suggested that I create the `issueUrl` in the `data: self` object instead of the `computed` object, but when I try to add it to the list there as `issueUrl: <the ridiculously long string>,`
the page doesn't render at all, but also produces no errors in Terminal or the console. 

If I instead try `issueUrl: **return** <the ridiculously long string >,` it says "Failed to compile" in the browser and thinks there's an unexpected token. 

Do I need to adjust some `this` or `self` references somewhere to make it work from this section of the code? For reference, this is how the variable is used elsewhere in the code: 
```
<div>
        Feeling stuck? We'd love to hear what's confusing so we can improve
        this lesson. Please <a :href="issueUrl">share your questions and feedback</a>.
      </div>
```

(What's been committed is the working code before attempting this maneuver, not the failed maneuver.)

olizilla commented:
I'd just go with the `computed` property. I don't know vue, but the docs suggest that computed props are the place to things that you want to derive from other props, as we're doing here

https://vuejs.org/v2/guide/computed.html

terichadbourne commented:
This branch now also addresses issue #65, correcting a grammar issue in success messages.

terichadbourne commented:
@olizilla Would you have a minute tomorrow to test this updated `question-link` branch for me? Two new things to check for: 

1. You should now see the success message "Everything works!" in most lessons (previously "All works!") . _addresses issue #65_
![image](https://user-images.githubusercontent.com/19171465/47386535-e18b0880-d6da-11e8-8155-a584ad0b4fd8.png)

2. In the final lesson of each series (basics 3 and blog 7), when you correctly solve the problem you should see the success message "Great job! You've completed this series of lessons!" and a "More Workshops" button that takes you back to the list of workshops, rather than a "Next" button that takes you to a non-existent subsequent lesson (blank white screen). _addresses issues #64 & #38_
![image](https://user-images.githubusercontent.com/19171465/47386614-126b3d80-d6db-11e8-8111-cab8fb8aea8c.png)

olizilla commented on src/components/Lesson.vue:
```diff
@@ -95,6 +98,12 @@
+    <section class="indent-1 mb4 mw-900">
+      <div>
+        <p>Feeling stuck? We'd love to hear what's confusing so we can improve
+        this lesson. Please <a :href="issueUrl">share your questions and feedback</a>.</p>
```
This would be more friendly if it opens github in a new tab, as we don't currently preserve your state across page reloads... right now if you click on it, your answer is lost.

```html
<a :href="issueUrl" target="_blank">
```

olizilla commented on src/lessons/Basics/03.vue:
```diff
@@ -37,7 +38,7 @@ const validate = async (result, ipfs) => {
 
   if (result.value === 1 && result.remainderPath === '') {
-    return {success: 'Great job!'}
+    return {success: "Great job! You've completed this series of lessons!"}
```
minor, but we're using singlequotes `'` for strings in this project. The bigger issue is we should have CI checks and code linting as part of the PR process, but that's an issue for another PR!

olizilla commented on src/lessons/Blog/07.vue:
```diff
@@ -170,7 +171,7 @@ const validate = async (result, ipfs) => {
     return {fail: `Your function threw an error: ${err}.`}
   }
-  return {success: 'All works!'}
+  return {success: "Great job! You've completed this series of lessons!"}
```
single quotes again

terichadbourne commented:
Great suggestions, @olizilla! I'll fix those items. I really appreciate you clueing me in to the nuances of formatting choices we're trying to be consistent with in the code across the repo.

terichadbourne commented:
Actually, @olizilla, I just remembered why I used double quotes there, and it's because that string includes an apostrophe in `you've`.  Would you rather I put backticks (template string format) instead of double quotes on the outside of strings when that's the case?

olizilla commented:
@terichadbourne I don't think we have anything set up to auto deploy this code yet, so it won't appear on the live site until it's manually updated. There is an issue to fix that #18 and another to fix the CI testing setup #71

olizilla commented:
@mikeal can you take a look at the CI/CD stuff when you get well? also, get well sir!
